transcom/mymove
# MAIN - B-22138 - Duty Location Results Should Not Include ZIP Codes for PO Boxes

## [B-22138](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-22138)

INT PR: https://github.com/transcom/mymove/pull/14482

## Summary

This is to add `is_po_box` column to `re_us_post_regions` table to keep record of postal codes being a PO Box or not. Query for searching duty locations in duty location dropdown(customer & office share the same query) have been updated to filter out duty locations where `is_po_box` is true.

- added `is_po_box` column to `re_us_post_regions` table and updated existing rows that need to have `is_po_box` as true (default is false).
- added an index for `is_po_box`
- added `is_po_box` to `v_locations` view.
- updated duty location dropdown search query to filter out duty locations where `is_po_box = true`.


### How to test

1. While creating a move as a customer -> use `92137` when searching for duty locations and verify that exact match does not exist. This zip is a PO Box so it should not be available in the results. 
2. Use any other zip that is not a PO Box and verify the results show. Also search for `San Diego` and verify a result with `92137` does not show.
3. You can also verify using any other PO Box zip.
4. Check the office user side (Edit Orders) and verify the results are the same when searching for any duty locations.

## Screenshots
![image](https://github.com/user-attachments/assets/1c5bfb71-b961-4afa-a4b0-988d17d5c999)
![image](https://github.com/user-attachments/assets/6179d215-3296-40ac-b7a3-4f3bd7c379c6)
![image](https://github.com/user-attachments/assets/6bc088f6-9fb5-4584-8543-5a9df72c2cdf)
![image](https://github.com/user-attachments/assets/cc60531b-39f6-4003-a329-d4ec12c08818)
